source/slang/slang-lookup-glslstd450.cpp

```diff
@@ -0,0 +1,133 @@
+// Hash function for GLSLstd450
+
+#include "../core/slang-common.h"
+#include "../core/slang-string.h"
+#include "spirv/unified1/GLSL.std.450.h"
+
+
+namespace Slang
+{
+
+static const unsigned tableSalt[81] ={
```

Comment:
Maybe worth having a comment here that this is using a perfect hashing algorithm and the hash function is generated by lookup-generator, to prevent people from trying to modify this file directly.